django/contrib/admin/options.py

```diff
@@ -475,24 +475,25 @@ def lookup_allowed(self, lookup, value, request=None):
                 # Lookups on nonexistent fields are ok, since they're ignored
                 # later.
                 break
+            if not prev_field or (
+                prev_field.is_relation
+                and field not in model._meta.parents.values()
+                and field is not model._meta.auto_field
+                and (
+                    model._meta.auto_field is None
+                    or part not in getattr(prev_field, "to_fields", [])
+                )
+                and (field.is_relation or not getattr(field, "primary_key", None))
```

Comment:
I have a question about this: the `not getattr(field, "primary_key", None)`, what is it covering exactly? is it asking for whether there is primary_key field? or if the primaty_key field is None? what happens if the value of `primary_key` is falsy?

I change the code to be:
```suggestion
                and (field.is_relation or not field.primary_key)
```
and no test fails, so I think that we can either simplify the code or add tests ensuring the have the more robust guard here.

Reply:
`not getattr(field, "primary_key", None)` was checking for `not(hasattr(field, "primary_key") and field.primary_key)`. But I think this was not necessary as all fields inherit from `Field` class which already has `primary_key` attribute. The only exception I can think is of reverse relations and for that `field.is_relation` is already set to True.

Reply:
Thanks @Hisham-Pak, would you have time to extend the test case or add a new one with such case so we evaluate if we need to further tweak the `if` guard?

Reply:
I don't think we need any further tweaking. It already is looking good.